geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/GeodeRedisSetWithFunctions.java

```diff
@@ -40,30 +47,18 @@ public GeodeRedisSetWithFunctions(ByteArrayWrapper key,
   }
 
   public static void registerFunctions() {
-    FunctionService.registerFunction(new SaddFunction());
-    FunctionService.registerFunction(new SremFunction());
-    FunctionService.registerFunction(new SmembersFunction());
-    FunctionService.registerFunction(new SdelFunction());
+    FunctionService.registerFunction(new CommandFunction());
   }
 
   @Override
   public long sadd(ArrayList<ByteArrayWrapper> membersToAdd) {
-    ResultCollector<ArrayList<ByteArrayWrapper>, List<Long>> results = FunctionService
-        .onRegion(region)
-        .withFilter(Collections.singleton(key))
-        .setArguments(membersToAdd)
-        .execute(SaddFunction.ID);
-
+    ResultCollector<Object[], List<Long>> results = executeFunction(SADD, membersToAdd);
     return results.getResult().get(0);
```

Comment:
this line is duplicated in each function. I don't think the `resultcollector` abstraction really makes sense in our case, because we are really trying to just call this on one server and get back one result.  Can we short circuit this by just calling this  `results.getResult().get(0)` in `executeFunction()`

Reply:
srem is different but I think we could make it the same. And then you have the generic challenge of getting executeFunction to return a generic value that the caller can specify. This seems doable and I like having as much of the common code in executeFunction as possible.